### PR TITLE
THRIFT-4738: Fix typescript declaration generation

### PR DESCRIPTION
**Pull request by mustafa-cosar**
This fixes some problems with the generated `.d.ts` files

### CODE REVIEW

**Comment by mustafa-cosar** [comment]
@jeking3 There are C# related errors on the travis.

File: compiler/cpp/src/thrift/generate/t_js_generator.cc [hunk 1]
```diff
@@ -510,6 +511,26 @@ string t_js_generator::render_includes() {
   return result;
 }
 
+/**
+ * Renders all the imports necessary for including another Thrift program
```
Comment (thehcma) [inline compiler/cpp/src/thrift/generate/t_js_generator.cc]: this comment seems a bit strange: do you mean "renders all of the imports necessary to allow this file to be included by other files"?

File: compiler/cpp/src/thrift/generate/t_js_generator.cc [hunk 2]
```diff
@@ -510,6 +511,26 @@ string t_js_generator::render_includes() {
...
+ */
+string t_js_generator::render_ts_includes() {
+  string result = "";
+
+  if (gen_node_) {
```
Comment (thehcma) [inline compiler/cpp/src/thrift/generate/t_js_generator.cc]: flip the test and unindent

File: compiler/cpp/src/thrift/generate/t_js_generator.cc (hunk 1 above)
Comment (mustafa-cosar) [reply compiler/cpp/src/thrift/generate/t_js_generator.cc]: No, assume there are 2 thrift files `A.thrift` and `B.thrift`.
B includes A: `include "A.thrift"`
This outputs to `B.d.ts` the following:
```ts
import A_ttypes = require('./A_types');
```

Comment (mustafa-cosar) [reply compiler/cpp/src/thrift/generate/t_js_generator.cc]: I think a single Thrift file is considered a Thrift program when generating code.

**Comment by jeking3** [comment]
Please rebase on master to clear out the CI issue - I just fixed it.

File: compiler/cpp/src/thrift/generate/t_js_generator.cc (hunk 2 above)
Comment (mustafa-cosar) [reply compiler/cpp/src/thrift/generate/t_js_generator.cc]: fixed

File: compiler/cpp/src/thrift/generate/t_js_generator.cc [hunk 3]
```diff
@@ -510,6 +511,26 @@ string t_js_generator::render_includes() {
   return result;
 }
 
+/**
+ * Renders all the imports necessary for including another Thrift program
+ */
+string t_js_generator::render_ts_includes() {
+  string result = "";
```
[the remark this answers is not in the document]
Comment (mustafa-cosar) [reply compiler/cpp/src/thrift/generate/t_js_generator.cc]: fixed

**Comment by mustafa-cosar** [comment]
I rebased on master

**Pull request merged**